luarules/gadgets/team_transfer/pipeline.lua

```diff
@@ -0,0 +1,219 @@
+		for i = 1, #hs do
+			local r = hs[i](ctx.senderTeamId, ctx.receiverTeamId, ctx.resource, ctx.amount)
+			if r ~= nil then return r end
+		end
+		return true
+	elseif policyType == Definitions.PolicyType.UnitTransfer then
+		local hs = legacy.onAllowUnitTransfer
+		for i = 1, #hs do
+			local r = hs[i](ctx.unitID, ctx.unitDefID, ctx.fromTeamID, ctx.toTeamID, ctx.capture)
+			if r ~= nil then return r end
+		end
+		return true
+	elseif policyType == Definitions.PolicyType.Command then
+		local hs = legacy.onAllowCommand
+		for i = 1, #hs do
+			local r = hs[i](ctx.unitID, ctx.unitDefID, ctx.unitTeam, ctx.cmdID, ctx.cmdParams, ctx.cmdOptions, ctx.cmdTag, ctx.synced)
+			if r ~= nil then return r end
+		end
+		return true
+	end
+end
+
+function Pipeline.RunAllowResourceTransfer(senderTeamId, receiverTeamId, resourceType, amount)
+	local resourceName = Resources.NormalizeResourceName(resourceType)
+	local maxShare = 0
+	local receiverCur = 0
+	if resourceName == 'metal' or resourceName == 'energy' then
+		maxShare, receiverCur = Resources.ComputeMaxShare(receiverTeamId, resourceName)
+	end
+	local clampedAmount = math.min(math.max(amount, 0), maxShare)
+	local ctx = {
+		type = Definitions.PolicyType.ResourceTransfer,
+		senderTeamId = senderTeamId,
+		receiverTeamId = receiverTeamId,
+		resource = resourceName,
+		amount = amount,
+		amountClamped = clampedAmount,
+		maxShare = maxShare,
+		receiverCur = receiverCur,
+		cumulativeMetal = Resources.GetCumulativeMetalSent(senderTeamId),
+		areAlliedTeams = Spring.AreTeamsAllied(senderTeamId, receiverTeamId),
+		isCheatingEnabled = Spring.IsCheatingEnabled(),
+		senderIsNonPlayer = isNonPlayerTeam(senderTeamId),
+		receiverIsNonPlayer = isNonPlayerTeam(receiverTeamId),
+	}
+
+	local res = evaluatePolicies(Definitions.PolicyType.ResourceTransfer, ctx)
+	if type(res) == "table" then
+		if res.applyTransfer then
+			local sent = res.applyTransfer.sent or 0
+			local received = res.applyTransfer.received or 0
+			Spring.SetTeamResource(receiverTeamId, resourceName, receiverCur + received)
+			local sCur = select(1, Spring.GetTeamResources(senderTeamId, resourceName))
+			Spring.SetTeamResource(senderTeamId, resourceName, sCur - sent)
+			if resourceName == 'metal' and res.applyTransfer.updateCumulativeMetal then
+				local newCum = Resources.AddCumulativeMetalSent(senderTeamId, sent)
+				Spring.SetTeamRulesParam(senderTeamId, "metal_share_cumulative_sent", newCum)
+			end
+			if res.expose then
+				if res.expose.threshold ~= nil then
+					Spring.SetTeamRulesParam(senderTeamId, "metal_share_threshold", res.expose.threshold)
+				end
+				if res.expose.taxRate ~= nil then
+					Spring.SetTeamRulesParam(senderTeamId, "resource_share_tax_rate", res.expose.taxRate)
+				end
+			end
+			return false
+		end
+		if res.allow ~= nil then
+			return res.allow
+		end
+		if res.deny ~= nil then
+			return not res.deny
+		end
+	elseif type(res) == "boolean" then
+		return res
+	end
+
+	return true
+end
+
```

Comment:
This was something that existed solely in the unit market, which no longer exists, so can be removed. (Pretty sure).